src/pages/UserPages/UserWallet/Wallet.jsx

```diff
@@ -0,0 +1,117 @@
+
+
+    const[showTopUp, setShowTopUp] = useState(false)
+    const[receiptNeeded, setReceiptNeeded] = useState(true)
+
+    const{ buttonDisable } = useSelector(state => ({
+        buttonDisable : state.buttonState.buttonDisable
+    }), shallowEqual)
+
+    const dispatch = useDispatch()
+
+    const creditFunc = () => {
+        setShowTopUp(true)
+        setReceiptNeeded(true)
+    }
+
+    const debitFunc = () => {
+        setShowTopUp(true)
+        setReceiptNeeded(false)
+    }
+
+
+    const validationSchema = Yup.object().shape({
+        amount: receiptNeeded ? Yup.number().min(1000).max(1000000).required().label("Credit Amount") : Yup.number().min(1000).max(wallet.available).required().label("Debit Amount"),
+        receipt: receiptNeeded && Yup.string().required().label("Receipt/Teller Image")
+    })
+
+
+    const submitForm = ({ amount, receipt }) => {
+
+        const transactionData = {
+            txType : receiptNeeded ? "credit" : "debit",
+            uid,
+            amount,
+            receipt : receiptNeeded ? receipt : "",
+            walletTransactions
+        }
+
+        const newWallet = {
+            pendingCredit : receiptNeeded ? wallet.pendingCredit + amount : wallet.pendingCredit,
+            pendingDebit : receiptNeeded ? wallet.pendingDebit : wallet.pendingDebit + amount,
+            available : receiptNeeded ? wallet.available : wallet.available - amount
+        }
+
+        dispatch(sendTransaction(transactionData))
+        dispatch(disableButton())
+        dispatch(updateWallet(newWallet))
+    }
+
+    return(
+        <>
+
+
+
+            <BaseModal
+                title={ receiptNeeded ? `Top Up Wallet` : `Wallet Withdrawal`}
+                show={showTopUp}
+                close={() => setShowTopUp(false)}
+                closeText="Close"
+                size="md"
+            >
+
+                <Form
+                    initialValues={{ amount : 1000, receipt : "" }}
+                    onSubmit={submitForm}
+                    validationSchema={validationSchema}
+                >
+
+                    <FormField
+                        name="amount"
+                        type="number"
+                        placeholder="0.00"
+                    />
+
+                    {receiptNeeded && <FormUpload 
+                        name="receipt"
+                        sizeLimit={204800}
+                        fileDestination={`receipt/${uid}`}
+                        fileType={['image/jpg', 'image/png', 'image/jpeg']}
+                    />}
+
+                    <SubmitButton title="Send For Approval" disable={buttonDisable} />
+
+
+                </Form>
+            </BaseModal>
+
+
+            <div>Available : {thousands_separators(wallet.available.toFixed(2))}</div>
+            <div>Pending Credit : {thousands_separators(wallet.pendingCredit.toFixed(2))}</div>
+            <div>Pending Debit : {thousands_separators(wallet.pendingDebit.toFixed(2))}</div>
```

Comment:
I'm not having access to the `pendingDebit` property on wallet and thus breaking the app

I'm getting ```{pending: 0, available: 0}```

Reply:
For this I can see that the `pendingCredit` and `pendingDebit` vaariables are being saved, so that means that this component can not (or should not according to your implementation) should not be viewed  if a user has not made any transaction. If this is intentional then if a user tht hasn't made any transaction comes to this route the user should then be redirected.

Reply:
Yeah...I updated the database structure... so you might have to update yours as well through the Firebase Console....

Reply:
I don't get your second comment, but this is a Wallet component. It will always be seen by the user. If the user doesn't have any pending Debits or Credits, It'll just read `0.00`

Reply:
Oooh ok, I didn't notice the change in the dB structure, thanks.